### PR TITLE
Merge demol6798:develop into base:develop - Reverse Sentence Algorithm and Bug Fix in Split Algorithm

### PR DESCRIPTION
**Pull request by demol6798**
Changed the wording of commit messages and fixed indentation as referred to by linked style guides.
Unfortunately, most commits contain identical changes. Wiped previous repo a couple of times, expected commits to be deleted as well. Guess not.

### CODE REVIEW

File: C/Algorithms/String/reverseSentence.c [hunk 1]
```diff
@@ -6,17 +6,20 @@
...
-    char **sentenceWords;
+    char *sentenceWords[20]= { NULL };	//sentenceWords = total number of individual words
+    static char *reverseSentence[20];	//reverseSentence = max num of individual words
+    int sentenceWordsLength = sizeof(sentenceWords) / sizeof(char*);
+
     split(sentence, ' ', sentenceWords);
 
-    char *reverseSentence = "";
-    int sentenceWordsLength = sizeof(sentenceWords) / sizeof(sentenceWords[0]);
+    for (int j = sentenceWordsLength - 1, i = 0;j > -1;j--, i++) {
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/reverseSentence.c]: Unnecessary blank line

File: C/Algorithms/String/reverseSentence.c [hunk 2]
```diff
@@ -6,17 +6,20 @@
...
 
-    for (int j = sentenceWordsLength - 1;j > -1;j--) {
-        strcat(reverseSentence, strcat(sentenceWords[j], ' '));
+	if(sentenceWords[j])
+	    reverseSentence[i] = strcat(sentenceWords[j], " ");
+
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/reverseSentence.c]: Unnecessary blank line

File: C/Algorithms/String/reverseSentence.c [hunk 3]
```diff
@@ -6,17 +6,20 @@
...
     split(sentence, ' ', sentenceWords);
 
-    char *reverseSentence = "";
-    int sentenceWordsLength = sizeof(sentenceWords) / sizeof(sentenceWords[0]);
+    for (int j = sentenceWordsLength - 1, i = 0;j > -1;j--, i++) {
 
-    for (int j = sentenceWordsLength - 1;j > -1;j--) {
-        strcat(reverseSentence, strcat(sentenceWords[j], ' '));
+	if(sentenceWords[j])
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/reverseSentence.c]: Always use { } for conditional statements (if-else) and loops (for, while, do-while) even if they include single statement only

File: C/Algorithms/String/split.c [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include <string.h>
+#include "string.h"
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Include standard C libraries with <> while include our written C files with " ". Include <stdlib.h> library for malloc() function.

File: C/Algorithms/String/split.c [hunk 5]
```diff
@@ -10,31 +10,31 @@
...
 {
-    int i = 0, j = 0, k = 0;
-    while (string[i] != '\0') {
-        if (string[i] == delimiter) {
-            char *subString;
-
-            for (int a = j, b = 0;a < i;a++, b++) {
-                subString[b] = string[a];
+    int i = -1, j = 0, k = 0;
+
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c [hunk 6]
```diff
@@ -10,31 +10,31 @@
...
-        if (string[i] == delimiter) {
-            char *subString;
-
-            for (int a = j, b = 0;a < i;a++, b++) {
-                subString[b] = string[a];
+    int i = -1, j = 0, k = 0;
+	
+    do {
+	i++;
+
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c [hunk 7]
```diff
@@ -10,31 +10,31 @@
...
+	if (string[i] == delimiter || string[i] == '\0') {
+	    char *subString = malloc(sizeof(char) * 20); //subString = max length of individual words
+
+	    for (int a = j, b = 0; a <= i; a++, b++) {
+
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c [hunk 8]
```diff
@@ -10,31 +10,31 @@
...
+	if (string[i] == delimiter || string[i] == '\0') {
+	    char *subString = malloc(sizeof(char) * 20); //subString = max length of individual words
+
+	    for (int a = j, b = 0; a <= i; a++, b++) {
+                
+		if (a == i) {
+		    subString[b] = '\0';
+		}
+		else {
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: else is written on the same line as the closing parenthesis of if statement ( } else { )

File: C/Algorithms/String/split.c [hunk 9]
```diff
@@ -10,31 +10,31 @@
...
+		    subString[b] = string[a];
+		}
+
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c [hunk 10]
```diff
@@ -10,31 +10,31 @@
...
+		else {
+		    subString[b] = string[a];
+		}
+		
             }
 
             words[k] = subString;
             k++;
             j = i + 1;
+
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c [hunk 11]
```diff
@@ -10,31 +10,31 @@
...
+		    subString[b] = string[a];
+		}
+		
             }
 
             words[k] = subString;
             k++;
             j = i + 1;
+			
         }
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/reverseSentence.c (hunk 1 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/reverseSentence.c]: Unnecessary blank line

File: C/Algorithms/String/reverseSentence.c [hunk 12]
```diff
@@ -6,17 +6,20 @@
...
+	    reverseSentence[i] = strcat(sentenceWords[j], " ");
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/reverseSentence.c]: Always use { } for conditional statements (if-else) and loops (for, while, do-while) even if they include single statement only

File: C/Algorithms/String/reverseSentence.c (hunk 2 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/reverseSentence.c]: Unnecessary blank line

File: C/Algorithms/String/split.c (hunk 4 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Include standard C libraries with <> while include our written C files with " ". Include <stdlib.h> library for malloc() function.

File: C/Algorithms/String/split.c (hunk 6 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c (hunk 7 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c (hunk 8 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: else is written on the same line as the closing parenthesis of if statement ( } else { )

File: C/Algorithms/String/split.c (hunk 9 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c (hunk 10 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c (hunk 11 above)
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c [hunk 13]
```diff
@@ -10,31 +10,31 @@
...
-
-    int stringLength = strlen(stringLength);
-    char *subString;
-
-    for (int a = j, b = 0;a < stringLength;a++, b++) {
-        subString[b] = string[a];
-    }
-
-    words[k] = subString;
+	} while (string[i] != '\0');
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Improve code indentation

File: C/Algorithms/String/split.c [hunk 14]
```diff
@@ -10,31 +10,31 @@
...
```
Comment (ArmenGabrielyan16) [inline C/Algorithms/String/split.c]: Unnecessary blank line

File: C/Algorithms/String/split.c (hunk 13 above)
Comment (demol6798) [inline C/Algorithms/String/split.c]: Please review and accept changes.

**Comment by demol6798** [comment]
Please review and accept changes.

**Pull request merged**